pycsw/ogc/api/records.py

```diff
@@ -1048,22 +1048,70 @@ def record2json(record, url, collection, stac_item=False):
+                'title': link['name'],
+                'href': link['url']
+                }
+            if link['protocol']:
+                if 'OGC:WMS' in link['protocol'].upper():
+                    link2['rel'] = 'map'
+                    link2['templated'] = 'true'
+                    # assumes link['url'] includes '&layers=...', else link['name'] contains layername(s) 
+                    link2['href'] = merge_qs(link['url'], { 
+                            'request': 'GetMap', 'service': 'WMS',   
+                            'width': '{width}', 'height': '{height}', 'bbox': '{bbox}'}, {
+                            'version': '1.3.0', 'crs': 'epsg:4326', 
+                            'layers': link['name'], 'format': 'image/png'})
+                    link2['variables'] = {
+                            'bbox': {
+                                'type': 'array', 
+                                'items': {'type': 'number', 'format': 'double'},
+                                'minItems': 4, 'maxItems': 4
+                                },
+                            'width': {'type': 'number', 'format': 'integer'},
+                            'height': {'type': 'number', 'format': 'integer'}
+                            }
+                    link2['type'] = 'image/png'
+                    
+                elif 'OGC:WMTS' in link['protocol'].upper():
+                    link2['rel'] = 'map'
+                    link2['templated'] = 'true'
+                    link2['href'] = merge_qs(link['url'], { 
+                        'service': 'WMTS', 'request': 'GetTile', 'version': '1.0.0',
+                        'TileMatrix': '{TileMatrix}', 'TileRow': '{TileRow}', 'TileCol': '{TileCol}'}, {
+                        'TileMatrixSet': 'default', 'Layer': link['name'], 
+                        'Style': 'default', 'Format': 'image/png'})
+                    link2['variables'] = { 
+                        'TileMatrix': {'type': 'number', 'format': 'integer'},
+                        'TileRow': {'type': 'number', 'format': 'integer'},
+                        'TileCol': {'type': 'number', 'format': 'integer'}
+                    }
+                    link2['type'] = 'image/png'
+                    
+                elif 'OGC:WFS' in link['protocol'].upper():   
+                    link2['rel'] = 'map'
+                    link2['templated'] = 'true'
+                    link2['href'] = merge_qs(link['url'], {
+                        'version': '2.0.0', 'request': 'GetFeature', 'service': 'WFS', 'count': '{count}'}, {
+                        'typenames': link['name'], 'srsName': 'urn:ogc:def:crs:EPSG::4326',
+                        'outputFormat':'application/gml+xml'})
+                    link2['variables'] = { 
+                        'count': {'type': 'number', 'format': 'integer'}}
+                    link2['type'] = 'application/gml+xml'
+
+                #elif 'OSGEO:TMS' in link['protocol'].upper():
```

Comment:
or better/also to use xyz (both not in catinterop yet, see https://github.com/OSGeo/Cat-Interop/issues/42)